### DISCUSSION TITLE
How is your experience in OSes other than Linux?

### DISCUSSION THREAD

**Discussion by gerardbosch** [kmonad/kmonad#412] [General]
I'm just curious if KMonad is usable at the same level in MacOS and Windows as it is on Linux. Is there any known or serious limitation/problem/... right now? Thanks! :)

**Comment by molleweide** [comment]
It works really well on macos.
I am startong it manually tho because I am to scared too run it as a laucnh daemon because my layout is so complicated. It runs perfectly on both intel and M1 macs.

**Reply by molleweide** [reply]
You have to build it from source on mac

**Comment by reegnz** [comment]
For me setting it up on MacOS was a terrible experience. No1:  no homebrew tap! I'd be fine with that if there would be some downloadable binary builds. I'd even be fine if I had to build from source if there'd be a stable commit that 'just works'. I was entertaining the idea of writing a Homebrew tap for kmonad (it's wouldn't be that difficult), but this is where the real obstacles started to show up:

Building with dext support was quite simple, but the `master` branch simply doesn't work on MacOS right now. I had to jump through some hoops which completely demotivated me to continue:
- having to build from a branch (`keycode-refactor`) that is out-of date by about a year, it's missing 90 commits from `master`.
- having to patch the code to not err out https://github.com/kmonad/kmonad/issues/605: 
- having to downgrade the Karabiner-Elements VirtualHIDDevice so kmonad manages to work, but that breaking karabiner-elements if I want to go back to that

Without a stable version-tracked code (properly tagged, and released, without having to monkey-patch manually: https://github.com/kmonad/kmonad/issues/614) that works on MacOS and not having to downgrade other dependencies either (only running on an ancient dext version: https://github.com/kmonad/kmonad/issues/502), I don't think I'd start building a distribution-build for it, or even suggest anyone on MacOS of trying to use it, as getting it to work is a massive time-sink.

The layers concept of kmonad is really nice, but with all that hassle it's not worth it on MacOS and I would recommend Karabiner+Goku for anybody on MacOS. The UX for Karabiner-Elements + Goku outweighs the massive inconvenience of setting up kmonad on Mac.

I was coming in with really high hopes for kmonad, but left with the impression that this is a linux-only tool advertised with a **very** alpha MacOS support. Misleading advertising.

**Reply by gerardbosch** [reply]
Thanks for your feedback! It would be really nice if someone or the authors of the project could write a small README guide for MacOs (and maybe Windows?) installation, build commands and/or any trick necessary along with the right commit/branch/tag version to make it work on those OSes.

KMonad became a core part of my laptop, and I'm afraid about changing from Linux to MacOs if I start to work somewhere where I cannot have a Linux computer.

BTW, @molleweide commented here it was working very well on MacOs one year ago: https://github.com/kmonad/kmonad/discussions/412#discussioncomment-1637007

**Reply by molleweide** [reply]
works perfectly on mac intel and silicon ships. now i haven't built kmo again in a while.

**Reply by reegnz** [reply]
@molleweide Would you care writing a guide? I don't see anything of my observations have changed since I wrote that comment.

**Comment by niklaas** [comment]
I would like to get a homebrew Formulae (plus Cask for the `dext`) working. I'm asking about the most feasible and convenient set-up for end-users [in the homebrew discussions](https://github.com/orgs/Homebrew/discussions/5015).

It's the first time I write a Formulae (and Cask) but tinkering with it, getting it to work seems straightforward. What I am wondering about is

1. how the get the dependency with `dext` working seamlessly
2. whether it's better to commit the Formulae (and Cask) to `homebrew-core` or create a tap for it.

If you have ideas or thoughts about this, I'd appreciate you sharing them.
